nemo/collections/asr/models/aed_multitask_models.py

```diff
@@ -656,105 +614,62 @@ def validation_step(self, batch, batch_idx, dataloader_idx=0, eval_mode="val"):
             transcript=input_ids,
             transcript_length=transcript_len,
         )
-
-        beam_hypotheses = self.decoding.decode_predictions_tensor(
-            encoder_hidden_states=enc_states,
-            encoder_input_mask=enc_mask,
-            decoder_input_ids=input_ids[:, : self.context_len_for_AR_decoding]
-            if self.context_len_for_AR_decoding > 0
-            else None,
-            return_hypotheses=False,
-        )[0]
-
         transf_loss = self.loss(log_probs=transf_log_probs, labels=labels)
-
-        ground_truths = [self.tokenizer.ids_to_text(sent) for sent in transcript.detach().cpu().tolist()]
-        translations = [hyp for hyp in beam_hypotheses]
-
         self.val_loss(loss=transf_loss, num_measurements=transf_log_probs.shape[0] * transf_log_probs.shape[1])
-
         output_dict = {
             f'{eval_mode}_loss': transf_loss,
-            'translations': [self.decoding.strip_special_tokens(t) for t in translations],
-            'ground_truths': [self.decoding.strip_special_tokens(g) for g in ground_truths],
         }
 
-        if type(self.trainer.val_dataloaders) == list and len(self.trainer.val_dataloaders) > 1:
-            self.validation_step_outputs[dataloader_idx].append(output_dict)
-        else:
-            self.validation_step_outputs.append(output_dict)
+        self.wer.update(
+            predictions=enc_states,
+            predictions_lengths=encoded_len,
+            targets=transcript,
+            targets_lengths=transcript_len,
+            predictions_mask=enc_mask,
+            input_ids=input_ids[:, : self.context_len_for_AR_decoding]
```

Comment:
Please refer to this PR where I removed `self.context_len_for_AR_decoding`, dataloader will now return properly formatted prompts for inference and doesn't need the text/answer field to be present.

https://github.com/NVIDIA/NeMo/pull/8432

Reply:
Done. Updated inference flag to just pass additional prompt tensor from dataloader.